stac_geoparquet/stac_geoparquet.py

```diff
@@ -46,6 +55,17 @@ def to_geodataframe(items: Sequence[dict[str, Any]]) -> geopandas.GeoDataFrame:
             if k in item2:
                 raise ValueError("k", k)
             item2[k] = v
+        if add_self_link:
+            self_href = None
+            for link in item["links"]:
+                if (
+                    link["rel"] == "self"
+                    and (not link["type"] or link["type"] in STAC_ITEM_TYPES)
```

Comment:
Can you explain the need for this second check? IIUC, it would filter out a `self` link with an incorrect type?

Reply:
I've seen OGC API implementations that provide multiple self links (e.g. text/html and application/json). I want to filter out everything that is obviously wrong :-)

Reply:
Ah, makes sense.